zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -1206,6 +1226,7 @@ public ProcessTxnResult processTxn(TxnHeader header, Record txn, boolean isSubTx
             case OpCode.closeSession:
                 long sessionId = header.getClientId();
                 if (txn != null) {
+                    ephemeralsByteSizeMap.remove(sessionId);
```

Comment:
if the user has disabled the check, will you still have an entry into the map<>?
this comment: / Only store sum of ephemeral node byte sizes if we're enforcing a limit

Reply:
The map would be empty if we are not enforcing a limit (aka limit is set to -1), so remove would return null and not do anything. 
Should it be guarded behind an if statement?

Reply:
No need to add condition. i re-checked that you are using concurrentHaspMap and how remove would work. i checked the API, we should be ok. thanks